elfloader-tool/src/arch-riscv/crt0.S

```diff
@@ -32,6 +32,7 @@ _start:
 
 /* a0 should have hart id, store it in s0 so as not to clobber from HSM calls */
   mv s0, a0
+  mv s2, a1
```

Comment:
Good catch, I have also created https://github.com/seL4/seL4_tools/issues/108 as a follow up. that we should take the passed device tree block more seriously - at least for debugging and porting that could be quite helpful. 
Could you add a comment here what this second parameters is, similar to the comment above? That improves the maintainability of the code a bit more.

Reply:
sure, have a look